sys/src/9/port/devramfs.c

```diff
@@ -0,0 +1,180 @@
+/*
+ * This file is part of the UCB release of Plan 9. It is subject to the license
+ * terms in the LICENSE file found in the top-level directory of this
+ * distribution and at http://akaros.cs.berkeley.edu/files/Plan9License. No
+ * part of the UCB release of Plan 9, including this file, may be copied,
+ * modified, propagated, or distributed except according to the terms contained
+ * in the LICENSE file.
+ */
+
+#include	"u.h"
+#include	"../port/lib.h"
+#include	"mem.h"
+#include	"dat.h"
+#include	"fns.h"
+#include	"../port/error.h"
+
+#define DIR_EXTEND_SIZE 10
+
+int memcpy(void* dst, const void* src, int len);
+
+struct Directory {
+	int inuse;
+	int allocated;
+	Dirtab ramdir[];
+};
+
+struct RamFile {
+	int length;
+	char data[];
+};
+
+static struct Directory *ramroot;
```

Comment:
There should not be just one of these. The reason is that each time there is an attach it can be a different name space. So it needs to be dynamically allocated in attach.